js/settings.js

```diff
@@ -101,3 +103,52 @@ Settings.prototype.onChanged_ = function(changes, areaName) {
   }
 };
 
+/**
+ * Enables all settings to be editable.
+ */
+Settings.prototype.enableAll = function() {
+  for (const [key, setting] of Object.entries(Settings.SETTINGS)) {
+    const widget = setting.widget;
+
+    if (widget === 'checkbox') {
+      const elem = document.getElementById(`setting-${key}-switch`);
+      elem.classList.remove('mdc-switch--disabled');
+      elem.querySelector('input').removeAttribute('disabled');
+    } else if (widget === 'number') {
+      const elem = document.getElementById(`setting-${key}`);
+      elem.removeAttribute('disabled');
+    } else if (widget === 'search') {
+      const elem = document.getElementById('search-input');
+      elem.removeAttribute('disabled');
+      const container = document.querySelector('.search-container');
+      container.classList.remove('disabled');
+    }
+    // Else it can't be disabled.
+  }
+}
+
+
+/**
+ * Locks a setting to particular value and prevents the user from changing it.
+ * Can only lock a setting with a checkbox or a number widget.
+ */
+Settings.prototype.disable = function(setting, value) {
+  this.set(setting, value);
+
+  const widget = Settings.SETTINGS[setting].widget;
+  if (widget === 'checkbox') {
```

Comment:
why not use switch?

Reply:
I was unsure if that was good style (i dunno why bit i thought having non trivial logic in switch statements was discouraged), happy to switch it over!

Reply:
up to you :)